crates/oxc_syntax/src/symbol.rs

```diff
@@ -45,6 +48,9 @@ impl SymbolFlags {
     pub fn is_variable(&self) -> bool {
         self.intersects(Self::Variable)
     }
+    pub fn is_type(&self) -> bool {
```

Comment:
See https://github.com/microsoft/TypeScript/blob/71c56c24b6b2939323cb032934c5d9a0d95c413c/src/compiler/types.ts#L5759

We need to setup this thing exactly as it defined in TypeScript

Reply:
Do you want me to implement all of the symbol table rules in 1 PR? I tried to keep it small for the time being - I can try to implement the whole thing this weekend though

Reply:
No, we just need to add as we go, but leave rooms when we add more symbols.
I should've add the reference for this symbol flags. My bad.